src/utils.ts

```diff
@@ -1,39 +1,57 @@
-const setPaddingBetweenNodes = ({
-  context,
-  problemNode,
-  firstNode,
-  message,
-}) => {
-  context.report({
-    node: problemNode,
-    message,
-    fix: function(fixer) {
-      return fixer.insertTextAfter(firstNode, '\n');
-    },
-  });
-};
+import padding from './rules/padding';
 
-const getLeftSibling = node => {
-  const siblings = node.parent.body;
-  const nodePosition = siblings.indexOf(node);
-  return siblings[nodePosition - 1];
-};
+type TokenIdentifier =
+  | '*'
+  | 'afterAll'
+  | 'afterEach'
+  | 'beforeAll'
+  | 'beforeEach'
+  | 'describe'
+  | 'expect'
+  | 'it'
+  | 'test';
+
+interface Option {
+  blankLine: 'always' | 'any' | 'never';
+  prev: TokenIdentifier | TokenIdentifier[];
+  next: TokenIdentifier | TokenIdentifier[];
+}
 
-const getStartLine = node => node && node.loc.start.line;
+interface RuleDefs {
+  [name: string]: Option[];
+}
 
-const getEndLine = node => node && node.loc.end.line;
+/**
+ * Extend the base padding rule to make a new rule
+ */
+export const makeRule = (options: Option[]) => {
+  return {
+    meta: {
+      type: 'layout',
+      fixable: 'whitespace',
+    },
+    // Wrap base padding create with a function that modifies RuleContext
+    // with the options for the new rule
+    create(context) {
+      // Copy the RuleContext and overwrite options; it's frozen and
+      // we can't set them directly.
+      const ctx = Object.create(context, { options: { value: options } });
+
+      // Freeze it again
+      Object.freeze(ctx);
 
-const shouldFixGap = (bottomNode, topNode) =>
-  getStartLine(topNode) - getEndLine(bottomNode) < 2;
+      // Call the original create method with new context
+      return padding.create(ctx);
+    },
+  };
+};
 
-export const padBefore = ({ context, node, beforeMessage }) => {
-  const leftSibling = getLeftSibling(node);
-  if (leftSibling && shouldFixGap(leftSibling, node)) {
-    setPaddingBetweenNodes({
-      context,
-      problemNode: node,
-      firstNode: leftSibling,
-      message: beforeMessage,
-    });
-  }
+/**
+ * Build a set of rules from a RuleDefs object
+ */
+export const makeRules = (ruleDefs: RuleDefs) => {
+  return Object.keys(ruleDefs).reduce(
+    (rules, key) => Object.assign(rules, { [key]: makeRule(ruleDefs[key]) }),
```

Comment:
Can you make this its own function defined above.

Reply:
I ended up clobbering this whole `makeRules` function in favor of building the exported rules object directly and calling `makeRule` for each config.

`makeRules` just wasn't worth the indirection.